packages/wallet/src/keystore-wallet.ts

```diff
@@ -0,0 +1,140 @@
+import type { AbstractAddress } from '@fuel-ts/interfaces';
+import crypto from 'crypto';
```

Comment:
Two things:
 - The `crypto` lib will work on Node but not inside the browser
 - The encrypt/decrypt implementations for the Browser will probably differ from the Node one

The `<repoRood>/packages/keystore` tried to create a `universal-crypto` strategy, but it is broken since it doesn't support ESM:
 - https://github.com/FuelLabs/fuels-ts/issues/909

We'll have to fix this [universal-crypto strategy] before this PR can continue.

Reply:
@arboleya Nice catch. I will need to think of another approach to solve this

Reply:
@Torres-ssf The [blocking PR](https://github.com/FuelLabs/fuels-ts/pull/1049) has been finished and is awaiting review.

The points I raised in the above comment remain valid and should guide the rest of your implementation. Let me know if it isn't clear, and we can review it together.

The crucial part here is how to validate things thoroughly since we're currently [not running](https://github.com/FuelLabs/fuels-ts/pull/1049/commits/276fb40d5a063f0ce62bab050e4b58cc6fbe3b48) tests inside the browser, which might require manual testing for the time being.

Reply:
@arboleya Thanks for all the hard work on #1049 :pray:

Reply:
@Torres-ssf is this PR ready for review now? If not, let's convert this to a draft until it is.